editor/store/reducer.js

```diff
@@ -759,6 +765,39 @@ export const reusableBlocks = combineReducers( {
 					},
 				};
 			}
+
+			case 'REMOVE_REUSABLE_BLOCK': {
+				const { id } = action;
+				return omit( state, id );
+			}
+		}
+
+		return state;
+	},
+
+	isFetching( state = {}, action ) {
+		switch ( action.type ) {
+			case 'FETCH_REUSABLE_BLOCKS': {
+				const { id } = action;
+				if ( ! id ) {
+					return state;
+				}
+
+				return {
+					...state,
+					[ id ]: true,
+				};
+			}
+
+			case 'FETCH_REUSABLE_BLOCKS_SUCCESS':
+			case 'FETCH_REUSABLE_BLOCKS_FAILURE': {
+				const { id } = action;
+				if ( ! id ) {
+					return state;
```

Comment:
In what situations would you expect this code path to be hit? Even if it was falsey, would there be harm in allowing `omit` to proceed? Aside from returning a new reference: But there's also an issue that if `id` is truthy but doesn't exist in the current set, we still return a new reference.

```js
omit( state, 'some-id-that-doesnt-exist' ) !== state
```

Reply:
You're right, this check isn't necessary. I've removed it in 92b6a18fc1f4b936358c4e0ddb8a9fc0957354aa.